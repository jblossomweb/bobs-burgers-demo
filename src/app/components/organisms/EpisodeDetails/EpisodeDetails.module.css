.content {
  @apply w-full animate-fadein;
}

.content h1 {
  @apply text-4xl font-bold mb-4;
}

.content h2 {
  @apply text-2xl font-bold my-4;
}

.content p {
  @apply my-2 text-base;
}

.content ul {
  @apply my-2 text-base;
}

.links {
  @apply my-8;
}
