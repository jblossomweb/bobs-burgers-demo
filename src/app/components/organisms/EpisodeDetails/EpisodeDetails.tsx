import React from 'react'
import { Link } from 'react-router-dom'
import type Episode from 'app/models/Episode'
import Button from 'app/components/atoms/Button'
import styles from './EpisodeDetails.module.css'

export interface Props {
  episode: Episode
}

const EpisodeDetails: React.FC<Props> = ({ episode }) => (
  <div className={styles.content}>
    <h1>{episode.name}</h1>
    <h3>
      S{episode.season}E{episode.episode}
    </h3>
    <p>{episode.airDate}</p>
    <ul>
      <li>Production Code: {episode.productionCode}</li>
      <li>Total Viewers: {episode.totalViewers}</li>
    </ul>
    <div className={styles.links}>
      <Link to="/episodes">
        <Button hollow>&lt; All Episodes</Button>
      </Link>
    </div>
  </div>
)

export default EpisodeDetails
