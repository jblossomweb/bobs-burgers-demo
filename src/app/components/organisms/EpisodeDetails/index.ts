export { default, type Props } from './EpisodeDetails'
