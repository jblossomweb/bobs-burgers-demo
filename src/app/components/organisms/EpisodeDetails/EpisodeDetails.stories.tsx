import React from 'react'
import type { ComponentStory, ComponentMeta } from '@storybook/react'
import mockEpisode from 'app/models/__mocks__/episode.json'

import EpisodeDetails, { type Props } from './EpisodeDetails'

export default {
  title: 'components/organisms/EpisodeDetails',
  component: EpisodeDetails,
  argTypes: {
    episode: { control: 'object' },
  },
} as ComponentMeta<typeof EpisodeDetails>

const Template: ComponentStory<typeof EpisodeDetails> = ({
  episode,
}: Props) => <EpisodeDetails episode={episode} />

export const Sample = Template.bind({})
Sample.args = {
  episode: mockEpisode,
}
