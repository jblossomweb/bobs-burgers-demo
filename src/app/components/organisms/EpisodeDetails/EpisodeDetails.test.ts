import storySnaps from 'test/utils/storySnaps'
import * as stories from './EpisodeDetails.stories'

describe('app/components/organisms/EpisodeDetails', () => {
  describe('stories', () => {
    storySnaps(stories)
  })
})
