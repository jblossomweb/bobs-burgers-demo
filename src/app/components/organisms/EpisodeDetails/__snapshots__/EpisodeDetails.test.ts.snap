// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/organisms/EpisodeDetails stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="content"
  >
    <h1>
      "Burger War"
    </h1>
    <h3>
      S1E10
    </h3>
    <p>
      April 10, 2011
    </p>
    <ul>
      <li>
        Production Code: 1ASA03
      </li>
      <li>
        Total Viewers: 4.00 Million
      </li>
    </ul>
    <div
      class="links"
    >
      <a
        href="/episodes"
      >
        <button
          class="button hollow"
          type="button"
        >
          &lt; All Episodes
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
